Source/Core/DolphinQt2/GameList/GameList.cpp

```diff
@@ -539,8 +554,40 @@ static bool CompressCB(const std::string& text, float percent, void* ptr)
 
 void GameList::OnHeaderViewChanged()
 {
-  Settings::GetQSettings().setValue(QStringLiteral("tableheader/state"),
-                                    m_list->horizontalHeader()->saveState());
+  // So here's the deal: Qt's way of resizing stuff around stretched columns sucks ass
+  // That's why instead of using Stretch, we'll just make resizable columns take all the available
+  // space ourselves!
+
+  int available_width = width() - style()->pixelMetric(QStyle::PM_ScrollBarExtent);
+  int previous_width = 0;
+
+  std::vector<int> candidate_columns;
+
+  // Iterate through all columns
+  for (int i = 0; i < GameListModel::NUM_COLS; i++)
+  {
+    if (m_list->isColumnHidden(i))
+      continue;
+
+    if (m_list->horizontalHeader()->sectionResizeMode(i) == QHeaderView::Fixed)
+    {
+      available_width -= m_list->columnWidth(i);
+    }
+    else
+    {
+      candidate_columns.push_back(i);
+      previous_width += m_list->columnWidth(i);
+    }
+  }
+
+  for (int column : candidate_columns)
+  {
+    int column_width = static_cast<int>(
+        std::max(5.f, std::ceil(available_width * (static_cast<float>(m_list->columnWidth(column)) /
+                                                   previous_width))));
```

Comment:
This needs to be handled differently somehow, as currently it causes the table to overflow the width by `candidate_columns.size() - 1` pixels.

Reply:
Using `std::round()` instead of `std::ceil()` actually works perfectly here.

Reply:
Alright, I have a reasonable enough solution for this, and it looks nicer than leaving a few pixels of empty space. It's not the cleanest thing but it works perfectly, and that's what matters right? Add the following code here, and also see the 2 comments above this one.

```c++
new_width += column_width;
if (new_width > available_width)
  column_width -= new_width - available_width;
```